tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TsFileMetaData.java

```diff
@@ -150,9 +150,11 @@ public static TsFileMetaData deserializeFrom(ByteBuffer buffer) throws IOExcepti
 
     if (ReadWriteIOUtils.readIsNull(buffer)) {
       fileMetaData.createdBy = ReadWriteIOUtils.readString(buffer);
-    }
-    fileMetaData.totalChunkNum = ReadWriteIOUtils.readInt(buffer);
-    fileMetaData.invalidChunkNum = ReadWriteIOUtils.readInt(buffer);
+    }  
+    // if using v0.8.0 TsFile, use 0 to represent missing fields
+    fileMetaData.totalChunkNum = buffer == null ? 0 : ReadWriteIOUtils.readInt(buffer);
+    fileMetaData.invalidChunkNum = buffer == null ? 0 : ReadWriteIOUtils.readInt(buffer);

```

Comment:
Could this buffer really be null?

Reply:
> Could this buffer really be null?


Did you mean I should change `buffer == null`to `!ReadWriteIOUtils.readIsNull(buffer)` ?

Reply:
> Could this buffer really be null?

Thank you. I've understood this point.